### PR TITLE
add bot flag and per channel chat configs

### PR DESCRIPTION
**Pull request by slugalisk**
adds a flag to streams so we can move bots into their own category without losing them in the live streams section. 

adds config fields for chat channel/service so users can choose a chat to show with their stream.

### CODE REVIEW

File: api/src/Streams.cpp [hunk 1]
```diff
@@ -285,7 +318,7 @@ void Streams::WriteStreamsJSON(
 }
 
 std::shared_ptr<Stream> Streams::Emplace(const Channel &channel) {
-  auto stream = std::make_shared<Stream>(db_, channel);
+  auto stream = std::make_shared<Stream>(db_, channel, channel);
```
Comment (KenanY) [inline api/src/Streams.cpp]: second channel here is the chat channel?

Comment (slugalisk) [reply api/src/Streams.cpp]: oh yeah, this should have been copied from the user here. nice catch 👍

**Comment by jbpratt78** [comment]
@slugalisk Can this be merged or is something needing worked on?

**Comment by slugalisk** [comment]
IIRC it was never tested. Unless it's blocking something this could be shelved and reimplemented in the new api...

**Comment by jbpratt78** [comment]
Shelved.

**Pull request closed without merging**